src/workerd/io/io-context.c++

```diff
@@ -318,6 +318,12 @@ kj::Maybe<IoOwn<kj::Promise<void>>> IoContext::waitForOutputLocksIfNecessaryIoOw
   });
 }
 
+bool IoContext::isOutputGateBroken() {
+  return actor.map([](Worker::Actor& actor) {
```

Comment:
nit: I think this will probably be simpler as a KJ_IF_MAYBE? Functional style flow control will generally be less optimizable than using language features.